backend-shared/util/fauxargocd/faux_argo_cd.go

```diff
@@ -111,3 +116,309 @@ type Backoff struct {
 	// MaxDuration is the maximum amount of time allowed for the backoff strategy
 	MaxDuration string `json:"maxDuration,omitempty" protobuf:"bytes,3,opt,name=maxDuration"`
 }
+
+type FauxApplicationStatus struct {
+	// Resources is a list of Kubernetes resources managed by this application
+	Resources []ResourceStatus `json:"resources,omitempty" protobuf:"bytes,1,opt,name=resources"`
+	// Sync contains information about the application's current sync status
+	Sync SyncStatus `json:"sync,omitempty" protobuf:"bytes,2,opt,name=sync"`
+	// Health contains information about the application's current health status
+	Health HealthStatus `json:"health,omitempty" protobuf:"bytes,3,opt,name=health"`
+	// Conditions is a list of currently observed application conditions
+	Conditions []ApplicationCondition `json:"conditions,omitempty" protobuf:"bytes,5,opt,name=conditions"`
+	// OperationState contains information about any ongoing operations, such as a sync
+	OperationState *OperationState `json:"operationState,omitempty" protobuf:"bytes,7,opt,name=operationState"`
+}
+
+// ResourceStatus holds the current sync and health status of a resource
+// TODO: describe members of this type
```

Comment:
Let's remove this TODO: it's from the original Argo CD file upon which this part of the PR is based (I'm sure they'll get around to doing it any day now :grin: )